coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/hierarchy/ProposalHierarchyServiceImpl.java

```diff
@@ -349,6 +349,18 @@ public void synchronizeAllChildren(DevelopmentProposal hierarchyProposal) throws
         prepareHierarchySync(hierarchyProposal);
         synchronizeAll(hierarchyProposal);
     }
+    
+    public void synchronizeAllChildrenBudgets(DevelopmentProposal hierarchyProposal) throws ProposalHierarchyException {
+    	prepareHierarchySync(hierarchyProposal);
+        finalizeHierarchySync(hierarchyProposal.getProposalDocument());
+
+        for (DevelopmentProposal childProposal : getHierarchyChildren(hierarchyProposal.getProposalNumber())) {
+            List<BudgetPeriod> oldBudgetPeriods = getOldBudgetPeriods(getHierarchyBudget(hierarchyProposal));
+            synchronizeChildBudget(hierarchyProposal, childProposal, oldBudgetPeriods);
+            dataObjectService.save(childProposal);
```

Comment:
Hmm...good question. We don't do that anywhere in the hierarchy service currently and things seem to work, but there are some strange persistence related bugs I haven't been able to track down in budget. I'll give that a try.